src/acreplace.rs

```diff
@@ -0,0 +1,38 @@
+use aho_corasick::AhoCorasickBuilder;
+use aho_corasick::AhoCorasick;
+use std::{
+    cell::RefCell,
+    collections::hash_map::HashMap
+};
+
+struct Replacements {
+    pub automaton: AhoCorasick,
+    pub replacements: Vec<String>
+}
+
+thread_local! {
+    static CREPLACE_MAP: RefCell<HashMap<String, Replacements>> = RefCell::new(HashMap::new());
+}
+
+byond_fn! { setup_acreplace(key, patternsjson, replacementsjson) {
+    let patterns: Vec<String> = serde_json::from_str(patternsjson.clone()).ok()?;
+    let replacements: Vec<String> = serde_json::from_str(replacementsjson.clone()).ok()?;
```

Comment:
```suggestion
byond_fn! { setup_acreplace(key, patterns_json, replacements_json) {
    let patterns: Vec<String> = serde_json::from_str(patterns_json.clone()).ok()?;
    let replacements: Vec<String> = serde_json::from_str(replacements_json.clone()).ok()?;
```

On that note, why are you using `.clone()` instead of just `&`?